From: Matthias Clasen <mclasen@redhat.com>
Date: Mon, 29 Mar 2021 20:48:15 -0400
Subject: label: Fix reacting to screen resolution changes

Labels were not reacting when the screen resolution
changes, as e.g. when turning "Large Text" on of off.

Origin: upstream, 3.24.29, commit:316aa85b8dadff8c1b621cb179e43aba0d232d6a
---
 gtk/gtklabel.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/gtk/gtklabel.c b/gtk/gtklabel.c
index 4bb92fd..bd5c397 100644
--- a/gtk/gtklabel.c
+++ b/gtk/gtklabel.c
@@ -4228,6 +4228,12 @@ gtk_label_style_updated (GtkWidget *widget)
   context = gtk_widget_get_style_context (widget);
   change = gtk_style_context_get_change (context);
 
+  if (change == NULL || gtk_css_style_change_affects (change, GTK_CSS_AFFECTS_FONT))
+    {
+      gtk_label_clear_layout (GTK_LABEL (widget));
+      gtk_widget_queue_resize (label);
+    }
+
   if (change == NULL || gtk_css_style_change_affects (change, GTK_CSS_AFFECTS_TEXT_ATTRS) ||
       (priv->select_info && priv->select_info->links))
     gtk_label_update_layout_attributes (label);
